src/v/crypto/tests/BUILD

```diff
@@ -84,3 +87,141 @@ redpanda_cc_gtest(
         "@googletest//:gtest",
     ],
 )
+
+expand_template(
+    name = "openssl_conf",
+    out = "openssl_conf.cnf",
+    substitutions = {
+        "@REDPANDA_DEPS_INSTALL_DIR@": "src/v/crypto/tests",
+    },
+    template = "test_openssl_conf.cnf.in",
+)
+
+copy_gen_file(
+    name = "fipsmodule_cnf",
+    dest = "etc/ssl/fipsmodule.cnf",
+    path = "ssl/fipsmodule.cnf",
+    source = "@openssl-fips",
+    deps = ["@openssl-fips"],
+)
+
+select_file(
+    name = "fipsmodule_module",
+    srcs = "@openssl-fips",
+    subpath = "lib/ossl-modules/fips.so",
+)
+
+redpanda_cc_bench(
+    name = "crypto_bench",
+    timeout = "short",
+    srcs = ["crypto_bench.cc"],
+    data = [
+        ":fipsmodule_cnf",
+        ":fipsmodule_module",
+        ":openssl_conf",
+    ],
+    env = {
+        "OPENSSL_CONF": "$(rootpath :openssl_conf)",
+        "MODULE_DIR": "src/v/crypto/tests/lib/ossl-modules",
+    },
+    deps = [
+        "//src/v/crypto",
+        "//src/v/random:generators",
+        "//src/v/ssx:thread_worker",
+        "@openssl-fips",
+        "@seastar",
+        "@seastar//:benchmark",
+    ],
+)
+
+redpanda_cc_bench(
+    name = "crypto_bench_fips",
+    timeout = "short",
+    srcs = ["crypto_bench.cc"],
+    data = [
+        ":fipsmodule_cnf",
+        ":fipsmodule_module",
+        ":openssl_conf",
+    ],
+    env = {
+        "OPENSSL_CONF": "$(rootpath :openssl_conf)",
+        "MODULE_DIR": "src/v/crypto/tests/lib/ossl-modules",
+        "PERF_FIPS_MODE": "1",
+    },
+    deps = [
+        "//src/v/crypto",
+        "//src/v/random:generators",
+        "//src/v/ssx:thread_worker",
+        "@openssl",
+        "@seastar",
+        "@seastar//:benchmark",
+    ],
+)
+
+redpanda_test_cc_library(
+    name = "ossl_context_service_test_base",
+    hdrs = [
+        "ossl_context_service_test_base.h",
+    ],
+    visibility = ["//visibility:private"],
+    deps = [
+        "//src/v/ssx:thread_worker",
+        "//src/v/test_utils:gtest",
+        "@googletest//:gtest",
+    ],
+)
+
+redpanda_cc_gtest(
+    name = "ossl_context_service_test",
+    timeout = "short",
+    srcs = [
+        "ossl_context_service_test.cc",
+    ],
+    cpu = 1,
+    data = [
+        ":fipsmodule_cnf",
+        ":fipsmodule_module",
+        ":openssl_conf",
+    ],
+    env = {
+        "OPENSSL_CONF": "$(rootpath :openssl_conf)",
+        "MODULE_DIR": "src/v/crypto/tests/lib/ossl-modules",
+    },
```

Comment:
We have these copied across quite a few targets.
Would it make sense to split them in a proxy library just to save a bit of space?
Or do we prefer being explicit about it?

Reply:
Remember that BUILD files are a subset of python so it's possible to declare a variable and reuse it.